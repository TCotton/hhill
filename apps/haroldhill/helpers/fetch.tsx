import {Entry, EntryCollection} from 'contentful'
import client from './contentful'
import mappedArticlesFn from './mappedArticlesFn'
import {IPagesFields} from "@hhill/types";
// TODO: come back and fix types
export const getArticles = async () => {
  return client
    .getEntries({
      content_type: 'chapters'
    })
    .then(
      (
        result
      ): {
        message?: string
        articles?: Record<string, unknown>
        mappedArticles?: Record<string, unknown>
      } => {
        const mappedArticles = mappedArticlesFn(result)
        return {
          message: 'ok',
          articles: mappedArticles.flatMap(
            (article: IPagesFields) => article.pages
          ),
          mappedArticles
        }
      }
    )
    .catch((err): { error?: string } => {
      return {
        error: err.name + err.message + err.stack.toString()
      }
    })
}

export const getSingleArticle = async (articleId: string) => {
  return client
    .getEntry(articleId.toString().split('-').pop() || '')
    .then((result): { message?: string; result?: Entry<unknown> } => {
      return {
        message: 'ok',
        result
      }
    })
    .catch((err): { message?: string } => {
      return {
        message: err.name + err.message + err.stack.toString()
      }
    })
}
