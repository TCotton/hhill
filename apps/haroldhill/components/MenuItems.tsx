import React, { useState, useEffect } from 'react'
import fetch from 'isomorphic-unfetch'
import Navigation from './Navigation'


const getArticles = async () => {
  const content = await fetch('http://localhost:3000/api/allArticles')
  const articles = await content.json()
  return articles.result.items.map((article) => {
    return {
      title: article.fields.title,
      slug: article.fields.slug,
      pages: article.fields.pages.map((page) => {
        return {
          title: page.fields.title,
          contentRichText: page.fields.contentRichText,
          slug: page.fields.slug,
          parent: article.fields.title,
          id: page.sys.id
        }
      })
    }
  })
}

const MenuItems = () => {
  const [articles, setArticles] = useState(null)

  useEffect(() => {
    getArticles().then((articles) => {
      setArticles(articles)
    })
  }, [])
  return <>{articles && <Navigation articles={articles} />}</>
}

export default MenuItems
