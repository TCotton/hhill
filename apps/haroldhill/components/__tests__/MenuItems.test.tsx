import React from 'react'
import { render, waitFor } from '@testing-library/react'
import fetch from 'isomorphic-unfetch'
import MenuItems from '../MenuItems'

jest.mock('isomorphic-unfetch', () =>
  jest.fn(() =>
    Promise.resolve({
      json: () =>
        Promise.resolve({
          result: {
            items: [
              {
                fields: {
                  title: 'Article 1',
                  slug: 'article-1',
                  pages: [
                    {
                      fields: {
                        title: 'Page 1.1',
                        contentRichText: 'Page 1.1 content',
                        slug: 'page-1.1',
                        parent: 'Article 1',
                        id: '12345'
                      },
                      sys: {
                        id: '12345'
                      }
                    }
                  ]
                }
              }
            ]
          }
        })
    })
  )
)

describe('MenuItems component', () => {
  it('renders the Navigation component with articles data', async () => {
    const { findByText } = render(<MenuItems />)
    expect(await findByText(/Article 1/i)).toBeInTheDocument()
    expect(await findByText('Page 1.1')).toBeInTheDocument()
    expect(await findByText('Page 1.1 content')).toBeInTheDocument()
    expect(fetch).toHaveBeenCalledWith('http://localhost:3000/api/allArticles')
  })
  it('should match snapshot', async () => {
    const { baseElement } = render(<MenuItems />)
    waitFor(() => {
      expect(baseElement).toMatchSnapshot()
    })
  })
})
