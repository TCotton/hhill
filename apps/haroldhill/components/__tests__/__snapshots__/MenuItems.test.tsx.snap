// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItems component should match snapshot 1`] = `
<body>
  <div />
</body>
`;
